### ISSUE TITLE
Parse error in/via zplg-shadow-off when restoring compdef / because of global alias

### ISSUE THREAD

**Issue by blueyed** [zdharma/zplugin#7]
I am getting the following error because of an `alias -g P='2>&1| pygmentize -l pytb'`:

```
-zplg-shadow-off:23: parse error: condition expected: "$opt"
-zplg-shadow-off:17: invalid function definition
```

As you said on IRC, disabling aliases would fix this.

**Comment by blueyed** [comment]
This fixes it for me, but it might be required in other places as well, of course.

```diff
diff --git i/zplugin.zsh w/zplugin.zsh
index 3c8f174..4cd3762 100644
--- i/zplugin.zsh
+++ w/zplugin.zsh
@@ -682,6 +682,7 @@ ZPLG_ZLE_HOOKS_LIST=(
 
 # Shadowing off completely for a given mode "load", "light" or "compdef"
 -zplg-shadow-off() {
+    setopt localoptions noaliases
     local mode="$1"
 
     # Disable shadowing only once
```

**Comment by psprint** [comment]
Thanks for the code! I've commited fix, added the setopt also for -zplg-shadow-on and to Zplugin in general.

**Comment by grimsleydl** [comment]
I believe this is causing issues with `zplg-load-snippet()` as well.

Unless I set noalias in this function: https://github.com/zdharma/zplugin/blob/039373af0ed625ed82d8fa299714b081528542ce/zplugin.zsh#L935 I get the following on starting a new shell:

    -zplg-load-snippet:23: parse error: condition expected: "$opt"
    -zplg-load-snippet:79: invalid function definition


Here's my .zshrc:

```
source './.zplugin/bin/zplugin.zsh'
autoload -Uz _zplugin
(( ${+_comps} )) && _comps[zplugin]=_zplugin
zplugin ice svn; zplugin snippet PZT::modules/fasd
zplugin load zsh-users/zsh-completions
zplugin snippet OMZ::lib/git.zsh

zplugin snippet OMZ::plugins/common-aliases/common-aliases.plugin.zsh
zplugin snippet OMZ::plugins/cp/cp.plugin.zsh
zplugin snippet OMZ::plugins/dotenv/dotenv.plugin.zsh
zplugin snippet OMZ::plugins/extract/extract.plugin.zsh
zplugin snippet OMZ::plugins/fancy-ctrl-z/fancy-ctrl-z.plugin.zsh
zplugin snippet OMZ::plugins/globalias/globalias.plugin.zsh
zplugin snippet OMZ::plugins/npm/npm.plugin.zsh
# zplugin snippet OMZ::plugins/perms/perms.plugin.zsh

zplugin light zdharma/zui
zplugin light zdharma/zplugin-crasis

zplugin load zsh-users/zsh-autosuggestions
zplugin load zdharma/history-search-multi-word
zplugin load zdharma/fast-syntax-highlighting
zplugin load djui/alias-tips
zplugin snippet ~/.fzf.zsh
# [ -f ~/.fzf.zsh ] && source ~/.fzf.zsh
```


I bisected my .zshrc until I found that the problem is in zplugin snippet OMZ::plugins/common-aliases/common-aliases.plugin.zsh:

Here are the aliases it defines--don't have the patience to narrow it down, but I'm assuming it's one of the single-letter aliases.
```
alias l='ls -lFh'     #size,show type,human readable
alias la='ls -lAFh'   #long list,show almost all,show type,human readable
alias lr='ls -tRFh'   #sorted by date,recursive,show type,human readable
alias lt='ls -ltFh'   #long list,sorted by date,show type,human readable
alias ll='ls -l'      #long list
alias ldot='ls -ld .*'
alias lS='ls -1FSsh'
alias lart='ls -1Fcart'
alias lrt='ls -1Fcrt'
alias zshrc='$EDITOR ~/.zshrc' # Quick access to the ~/.zshrc file
alias grep='grep --color'
alias sgrep='grep -R -n -H -C 5 --exclude-dir={.git,.svn,CVS} '
alias t='tail -f'
alias -g H='| head'
alias -g T='| tail'
alias -g G='| grep'
alias -g L="| less"
alias -g M="| most"
alias -g LL="2>&1 | less"
alias -g CA="2>&1 | cat -A"
alias -g NE="2> /dev/null"
alias -g NUL="> /dev/null 2>&1"
alias -g P="2>&1| pygmentize -l pytb"
alias dud='du -d 1 -h'
alias duf='du -sh *'
alias fd='find . -type d -name'
alias ff='find . -type f -name'
alias h='history'
alias hgrep="fc -El 0 | grep"
alias help='man'
alias p='ps -f'
alias sortnr='sort -n -r'
alias unexport='unset'
alias whereami=display_info
alias rm='rm -i'
alias cp='cp -i'
alias mv='mv -i'
alias -s pdf=acroread
alias -s ps=gv
alias -s dvi=xdvi
alias -s chm=xchm
alias -s djvu=djview
alias -s zip="unzip -l"
alias -s rar="unrar l"
alias -s tar="tar tf"
alias -s tar.gz="echo "
alias -s ace="unace l"
```

**Comment by psprint** [comment]
Thanks, it was `alias -g P=...`, which collided with Zsh `compdef` function, so it's like before. I've added a patch that fixes this.

**Comment by pecigonzalo** [comment]
Could it be we have a regression on this? Im having this exact issue

**Comment by blueyed** [comment]
@pecigonzalo 
Can you git-bisect it?

**Comment by pecigonzalo** [comment]
@blueyed I can confirm so far its present in `v2.05` the problem is, before that I get an error on
```
command not found: zpcompinit
```

So Im trying to work around that. Just to be clear, maybe I actually have to open another issue. Its not with `zplg-shadow-off` but rather with compdef.

Minimal code:
```
source "${HOME}/.zplugin/bin/zplugin.zsh"
zplugin snippet OMZ::plugins/common-aliases/common-aliases.plugin.zsh
zplugin ice wait"0" atinit"zpcompinit; zpcdreplay" lucid
zplugin load zsh-users/zsh-syntax-highlighting
```

**Comment by pecigonzalo** [comment]
I also confirm that `zplugin ice wait'1' lucid` before the `common-aliases` fixes it. To be honest, this is probably a bad escape and the fact that that plugin has single letter aliases, easy to cause problems.

**Comment by psprint** [comment]
I've added a fix for `zpcompinit` – it should use the -Uz options with `autoload` which disable aliases for when reading the function definition and also explicitly set the loading mode to Zsh (not Ksh). The `wait'1'` shouldn't be needed now :)

**Comment by pecigonzalo** [comment]
I just tested and it seems to work! Thanks <3

**Comment by hoshsadiq** [comment]
I think this issue has regressed. It disappears when I comment out the following alias, or add `wait'1'` to the snippet that loads the below (that is `OMZ::plugins/common-aliases/common-aliases.plugin.zsh`):
```
alias -g P="2>&1| pygmentize -l pytb"
```

**Comment by hoshsadiq** [comment]
Scratch that, seems adding `wait'1'` doesn't help, it just shows the errors a second later.

**Comment by psprint** [comment]
Thanks, I've fixed the issue.

**Comment by hoshsadiq** [comment]
Thanks
